src/Core/BasePlugin.php

```diff
@@ -321,4 +326,20 @@ public function events(EventManagerInterface $eventManager): EventManagerInterfa
     {
         return $eventManager;
     }
+
+    /**
+     * @return \Cake\Core\ContainerInterface|null
+     */
+    public function getContainer(): ?ContainerInterface
```

Comment:
Do we need this? I don't think we want to encourage the container spreading elsewhere.

Reply:
I too don't want to promote users directly using `$container->get()` by allowing easy access to the container instance.

Users should be accessing the container only to register services and we already have hook methods for that.